### ISSUE TITLE
Cannot edit documents "failed to read your document from storage"

### ISSUE THREAD

**Issue by iniOr** [nextcloud/richdocuments#841]
**Describe the bug**
Since a few days no documents can be edited, a prompt saying that the storage cannot be accessed pop all the time.

**To Reproduce**
1. Create a new document
2. Open it
3. try to save
4. "Failed to read document from storage...."

**Expected behavior**
Documents should be editable online.

**Client details:**
 - OS: Linux Mint
 - Browser Firefox
 - Version 73 x64
 - Device: Laptop

## Server details

**Operating system**: Debian 10

**Web server:** Apache

**Database:** Mariadb

**PHP version:** 7.3

**Nextcloud version:** 17.0.3 (homestorage encryption enabled)

**Version of the richdocuments app** No reference to richdocument found in apps.

**Version of Collabora Online** 3.5.2 (tried to downgrade to 3.5.0 but app is updated to 3.5.2 when i try to change the app on the apps folder (?))

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"Xljk29Wiea-GxOCxf@cJaQAAABg","level":3,"time":"2020-02-28T10:00:59+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"PHP","method":"GET","url":"\/apps\/richdocuments\/index?fileId=4426443&requesttoken=Mway%2FzjmzhN0M34mfIarJiSL4OOgt3zQPN7Ass4yTyE%3D%3Af2qdkFy1jzwxdBF1MrDDdW7jorPR01e8d7eN%2FrlIeBU%3D","message":"fread(): Length parameter must be greater than 0 at \/home\/web\/<domain.tld>\/apps\/files_external\/3rdparty\/icewind\/streams\/src\/Wrapper.php#91","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"Xljk3LTo8I8czhFBaoAqBAAAAAI","level":0,"time":"2020-02-28T10:01:00+00:00","remoteAddr":"172.17.0.2","user":"--","app":"richdocuments","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal?access_token=BEtn0EcmpishKHxUSSOvsVxCvTxGUraW&access_token_ttl=0","message":"Loaded WOPI Token record: {\"id\":\"981\",\"owner_uid\":\"orso\",\"editor_uid\":\"orso\",\"guest_displayname\":null,\"fileid\":\"4426443\",\"version\":\"0\",\"canwrite\":\"1\",\"server_host\":\"https:\/\/<domain.tld>\/\",\"token\":\"BEtn0EcmpishKHxUSSOvsVxCvTxGUraW\",\"expiry\":\"1582885859\",\"template_destination\":\"0\",\"hide_download\":\"0\",\"direct\":\"0\",\"is_remote_token\":\"0\",\"remote_server\":\"\",\"remote_server_token\":\"\",\"template_id\":\"0\"}.","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"Xljk3AA5yIojNmwemZjwyQAAAAU","level":0,"time":"2020-02-28T10:01:00+00:00","remoteAddr":"172.17.0.2","user":"--","app":"richdocuments","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal\/contents?access_token=BEtn0EcmpishKHxUSSOvsVxCvTxGUraW&access_token_ttl=0","message":"Loaded WOPI Token record: {\"id\":\"981\",\"owner_uid\":\"orso\",\"editor_uid\":\"orso\",\"guest_displayname\":null,\"fileid\":\"4426443\",\"version\":\"0\",\"canwrite\":\"1\",\"server_host\":\"https:\/\/<domain.tld>\/\",\"token\":\"BEtn0EcmpishKHxUSSOvsVxCvTxGUraW\",\"expiry\":\"1582885859\",\"template_destination\":\"0\",\"hide_download\":\"0\",\"direct\":\"0\",\"is_remote_token\":\"0\",\"remote_server\":\"\",\"remote_server_token\":\"\",\"template_id\":\"0\"}.","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"Xljk3AA5yIojNmwemZjwyQAAAAU","level":3,"time":"2020-02-28T10:01:00+00:00","remoteAddr":"172.17.0.2","user":"--","app":"PHP","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal\/contents?access_token=BEtn0EcmpishKHxUSSOvsVxCvTxGUraW&access_token_ttl=0","message":"fread(): Length parameter must be greater than 0 at \/home\/web\/<domain.tld>\/apps\/files_external\/3rdparty\/icewind\/streams\/src\/Wrapper.php#91","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"XljlUYQldg@VL4en0KuyyAAAAAE","level":0,"time":"2020-02-28T10:02:57+00:00","remoteAddr":"94.225.115.7","user":"--","app":"core","method":"GET","url":"\/apps\/theming\/styles?v=19","message":"SCSSCacher: \/apps\/theming\/css\/theming.scss compiled and successfully cached","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"Xljle7To8I8czhFBaoAqCQAAAAI","level":0,"time":"2020-02-28T10:03:39+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"maps","method":"GET","url":"\/apps\/files\/","message":"Could not find resource file \"\/apps\/maps\/js\/filetypes.js\"","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"XljlfLJOEbhlfIuKeokyFQAAAAY","level":0,"time":"2020-02-28T10:03:40+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"maps","method":"GET","url":"\/apps\/files\/","message":"Could not find resource file \"\/apps\/maps\/js\/filetypes.js\"","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"XljlfRgnbd3YVAQTSPKS@gAAABM","level":0,"time":"2020-02-28T10:03:41+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"ocr","method":"GET","url":"\/apps\/ocr\/api\/personal\/languages","message":"Get favorite languages for user: orso: ","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"XljlgLJOEbhlfIuKeokyGAAAAAY","level":3,"time":"2020-02-28T10:03:44+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"PHP","method":"GET","url":"\/apps\/richdocuments\/index?fileId=4426443&requesttoken=OOz3WXu38%2BiT3pRtDDv0BTIR41Zkl%2FOinxXXuOT7VNw%3D%3AdIDYNh%2FkssfWmfs%2BQg2cVnh5oQYV89jO1Hya9JOBY%2Bg%3D","message":"fread(): Length parameter must be greater than 0 at \/home\/web\/<domain.tld>\/apps\/files_external\/3rdparty\/icewind\/streams\/src\/Wrapper.php#91","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
{"reqId":"XljlgbBFPCcubGFDm2QnMQAAABQ","level":0,"time":"2020-02-28T10:03:45+00:00","remoteAddr":"172.17.0.2","user":"--","app":"richdocuments","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal?access_token=6ohIjukh60ArzFgrODT6PNTQY9FlCbzV&access_token_ttl=0","message":"Loaded WOPI Token record: {\"id\":\"982\",\"owner_uid\":\"orso\",\"editor_uid\":\"orso\",\"guest_displayname\":null,\"fileid\":\"4426443\",\"version\":\"0\",\"canwrite\":\"1\",\"server_host\":\"https:\/\/<domain.tld>\/\",\"token\":\"6ohIjukh60ArzFgrODT6PNTQY9FlCbzV\",\"expiry\":\"1582886024\",\"template_destination\":\"0\",\"hide_download\":\"0\",\"direct\":\"0\",\"is_remote_token\":\"0\",\"remote_server\":\"\",\"remote_server_token\":\"\",\"template_id\":\"0\"}.","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"XljlgXEx@Ri8UEtX9ng9GwAAABU","level":0,"time":"2020-02-28T10:03:45+00:00","remoteAddr":"172.17.0.2","user":"--","app":"richdocuments","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal\/contents?access_token=6ohIjukh60ArzFgrODT6PNTQY9FlCbzV&access_token_ttl=0","message":"Loaded WOPI Token record: {\"id\":\"982\",\"owner_uid\":\"orso\",\"editor_uid\":\"orso\",\"guest_displayname\":null,\"fileid\":\"4426443\",\"version\":\"0\",\"canwrite\":\"1\",\"server_host\":\"https:\/\/<domain.tld>\/\",\"token\":\"6ohIjukh60ArzFgrODT6PNTQY9FlCbzV\",\"expiry\":\"1582886024\",\"template_destination\":\"0\",\"hide_download\":\"0\",\"direct\":\"0\",\"is_remote_token\":\"0\",\"remote_server\":\"\",\"remote_server_token\":\"\",\"template_id\":\"0\"}.","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"XljlgXEx@Ri8UEtX9ng9GwAAABU","level":3,"time":"2020-02-28T10:03:45+00:00","remoteAddr":"172.17.0.2","user":"--","app":"PHP","method":"GET","url":"\/index.php\/apps\/richdocuments\/wopi\/files\/4426443_oc1j22tq1wal\/contents?access_token=6ohIjukh60ArzFgrODT6PNTQY9FlCbzV&access_token_ttl=0","message":"fread(): Length parameter must be greater than 0 at \/home\/web\/<domain.tld>\/apps\/files_external\/3rdparty\/icewind\/streams\/src\/Wrapper.php#91","userAgent":"LOOLWSD WOPI Agent 4.2.0","version":"17.0.3.1"}
{"reqId":"XljmgrJOEbhlfIuKeokyLAAAAAY","level":0,"time":"2020-02-28T10:08:03+00:00","remoteAddr":"94.225.115.7","user":"orso","app":"core","method":"GET","url":"\/settings\/user","message":"SCSSCacher: \/apps\/data_request\/css\/style.scss compiled and successfully cached","userAgent":"Mozilla\/5.0 (Windows NT 10.0; rv:68.0) Gecko\/20100101 Firefox\/68.0","version":"17.0.3.1"}
```

</details>

Thanks in advance for any help.

**Comment by juliushaertl** [comment]
Could you share some details on what kind of storage the file is located?

**Comment by juliushaertl** [comment]
`fread(): Length parameter must be greater than 0` seems to be in common with https://github.com/nextcloud/richdocuments/issues/833

**Comment by milkomeda** [comment]
Same problem here. Downgrade of richdocuments app to 3.5.0 fixed the problem for me.

**Comment by iniOr** [comment]
> Could you share some details on what kind of storage the file is located?

Hi, the file was located on the "main" encrypted (default nextcloud encryption module) storage.
This storage is located on a local hard drive.

Since I was experiencing a lot of issues and incompatibility with encryption I finally decided to disable encryption and since then I can again edit documents using Collabora Online without any problem, I still need to test it more but in my case encryption seems to be the culprit here.

> Same problem here. Downgrade of richdocuments app to 3.5.0 fixed the problem for me.

I tried to downgrade (removing local addon folder, getting version 3.5.0 from GitHub but doing that made my instance gone automatically to maintenance mode, upgrade richdocuments to latest version so I had to find something else...
